apollos-church-api/src/data/ContentChannel.js

```diff
@@ -26,16 +26,37 @@ export class dataSource extends RESTDataSource {
     });
   }
 
-  getPopularChannel = () => this.getFromId('{"search":"Messages_production","name":"Popular"}'); // todo: "filters":"ministries:The%20Porch",
+  getPopularChannel = () => this.getFromId(JSON.stringify({
+    search: this.context.dataSources.Search.messagesIndex,
+    name: 'Popular',
+    filters: 'ministries:"The Porch"',
+  }));
+
   getMessagesChannel = () => this.getFromId('https://media.watermark.org/api/v1/messages'); // todo
   getBlogChannel = () => this.getFromId('https://di0v2frwtdqnv.cloudfront.net/api/v1/property/theporch-app'); // todo
   getSeriesChannel = () => this.getFromId('https://media.watermark.org/api/v1/series?filter[tag_id][]=4'); // todo
 
-  getRootChannels = () => ([
+  getTopicsChannels = async () => {
+    const indice = this.context.dataSources.Search.indice(this.context.dataSources.Search.messagesIndex);
+    const { facets: { topics = {} } = {} } = await indice.search({
+      query: '',
+      facets: ['topics'],
+      filters: 'ministries:"The Porch"',
+    });
+
+    return Object.keys(topics).map((name) => (this.getFromId(JSON.stringify({
+      search: this.context.dataSources.Search.messagesIndex,
+      name,
+      filters: `ministries:"The Porch" AND topics:"${name}"`,
+    }))));
+  };
+
+  getRootChannels = async () => ([
     this.getPopularChannel(),
     this.getSeriesChannel(),
-    this.getMessagesChannel(),
+    // this.getMessagesChannel(),
```

Comment:
Question: Do we need this in the future?

Reply:
🤷‍♂️they don't want it right now